analysis/report.ipynb

```diff
@@ -0,0 +1,181 @@
+    "| First-ever positive or negative SARS-CoV2 test           | SGSS                             | `SGSS_Positive`, `SGSS_Negative`                   |\n",
+    "| A&E attendances                                          | SUS Emergency Care               | `EC`                                               |\n",
+    "| In-patient hospital admissions                           | SUS Admitted Patient Care Spells | `APCS`                                             |\n",
+    "| Out-patient hospital appointments                        | SUS                              | `OPA`                                              |\n",
+    "| Covid-related ICU admissions                             | ICNARC                           | `ICNARC`                                           |\n",
+    "| Covid-related in-hospital deaths                         | CPNS                             | `CPNS`                                             |\n",
+    "| COVID-19 Infection Survey                                | ONS                              | `ONS_CIS`                                          |\n",
+    "| All-cause registered deaths                              | ONS                              | `ONS_Deaths`                                       |\n",
+    "| High cost drugs                                          |                                  | `HighCostDrugs`                                    |\n",
+    "| Unique Property Reference Number<sup>1</sup>             |                                  | `UPRN`                                             |\n",
+    "| Master Patient Index                                     |                                  | `MPI`                                              |\n",
+    "| Health and Social Care Worker identification<sup>2</sup> |                                  | `HealthCareWorker`                                 |\n",
+    "\n",
+    "<sup>1</sup>Used for deriving household variables<br>\n",
+    "<sup>2</sup>Collected at the point of vaccination\n",
+    "\n",
+    "Some of these tables are accompanied by additional tables.\n",
+    "For example, the `OPA` table contains the core out-patient appointment event data, and is accompanied by the `OPA_Cost`, `OPA_Diag`, and `OPA_Proc` tables.\n",
+    "\n",
+    "### Report Run Date"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Markdown(\n",
+    "    f\"\"\"\n",
+    "This report was run on {date.today():%Y-%m-%d}.\n",
+    "It reflects the state of the database on this date.\n",
+    "\"\"\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Table Names by Data Source"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rows[[\"DataSource\", \"TableName\"]].drop_duplicates()"
```

Comment:
ah I see, here we drop the duplicates. here is just a thought that occurred to me, maybe we even spoke about it before:
 
do you think it would be useful to do this in a separate action, e.g., `tidy-data` that drops duplicates (and maybe does other data cleaning)?

Reply:
As it happens, I wrote a separate action for tidying data. Having done so, I realized I could move much of it to SQL (filtering, ordering) and was left with an action and tests that called `drop_duplicates` on a data frame 🙂 So, I think in general yes, but here no.